File: Python/Lamp3/Lamp.py
'''
Isaac Van Meter
Homework D-1
2/7/23
This is my own original work.
'''
class Lamp:

    __isOn = None

    def __init__(self, *args):
        if len(args) == 0:
            self.__isOn = False
        elif len(args) == 1:
            self.__isOn = args[0]
        else:
            raise Exception("Invalid Constuctor")
    

    def __str__(self):
        if self.__isOn:
            out = "{:3s} ".format("On")
        else:
            out = "{:3s} ".format("Off")
        return out

    def turnOn(self):
        self.__isOn = True

    def turnOff(self):
        self.__isOn = False

    def flip(self):
        self.__isOn = not self.__isOn

    def setLamp(self, isOn):
        self.__isOn = isOn

    def __eq__(self, other):
        return self.__isOn == other.__isOn

File: Java/Homework/Lamp.java

public class Lamp {

    private boolean isOn;

    public Lamp(){
        this.isOn = false;
    }

    public Lamp(boolean isOn){
        this.isOn = isOn;
    }

    public Lamp turnOn(){
        this.isOn = true;
        return this;
    }

    public Lamp turnOff(){
        this.isOn = false;
        return this;
    }

    public Lamp flip(){
        this.isOn = !isOn;
        return this;
    }

    public boolean isOn(){
        return this.isOn;
    }

    public Lamp setLamp(boolean isOn){
        this.isOn = isOn;
        return this;
    }

    @Override
    public String toString(){
        String out;
        if (this.isOn)
            out = String.format("%3s","ON");
        else
            out = String.format("%3s", "OFF");
        return out;
    }

    @Override
    public boolean equals(Object obj){
        Lamp other = (Lamp) obj;
        if (isOn != other.isOn){
            return false;
        }
        return true;
    }
}